src/background/background.ts

```diff
@@ -71,6 +71,22 @@ chrome.runtime.onConnect.addListener(port =>
             }
             break;
           }
+          case ExternalMethods.signatureRequest: {
+            void storePayload({
+              payload,
+              storageKey: StorageKey.signatureRequests,
+              port,
+            });
```

Comment:
In my current PR I've been considering how we can do this better. It seems odd to me that we store this payload with a key and then reread it. This makes it impossible to have multiple requests at the same time & leaves scope for a second request to overwrite the first.

Imo a better approach here is to store values with URL params. All we need is the `tabId` (and an id, coming in my refactor).

If we pass this to `popupCenter`,  we can read it with a custom hook like: 

```ts
// from my PR
export function useAccountRequestSearchParams() {
  const [searchParams] = useSearchParams();

  return useMemo(
    () => ({
      tabId: searchParams.get('tabId'),
      id: searchParams.get('id'),
    }),
    [searchParams]
  );
}
```

Then, when we want to send the message back, we can read the tabId from the URL and never have to store anything in localstorage.

Reply:
That's a good idea. I will try that, it should work better.

Reply:
This is fixed now. I used this strategy for both the `tabId` and the `origin` 👍